Clarify that both lifecycle and platform are programs (#158)
buildpacks/spec: platform.md

# Platform Interface Specification

> This document specifies how to process buildpack lifecycle steps to ensure compatibility
> between end-user tools.

Reviewer: ```suggestion
This document specifies the interface between a lifecycle program and a platform.
```
We can add "program," but this should cover the components we're defining an interface between:

Platform spec:
(platform <-> lifecycle) <-> buildpacks

Buildpack spec:
platform <-> (lifecycle <-> buildpacks)

Author: I didn't know there is a difference here. Maybe there are already some user stories about these differences? Stories that we could have started with in the first place to clarify problems that this design is meant to solve?

between end-user tools.

> A platform orchestrates a lifecycle to make buildpack functionality available to end-users such as application developers.

Reviewer: Could we keep this line?

Author: Explaining what a platform does instead of what platform is is exactly what makes this doc so confusing. There is an interface. The interface should be between something and something. Usually the interface is API, or CLI, or a protocol. When this context is clear, it could be said that `platform` is an command line app or server backend implemeting the API. But a straight mention that "some platform orchestrates some lifecyce" is like missing some links in the logic chain.

> Examples of tools implementing Platform Interface might include:

Reviewer: ```suggestion
Examples of platforms might include:
```
Not sure if a "cloud application platform" is clearly a tool

Author: Well, it is a not a platform either if it doesn't implement the interface.
